backend/src/graphql/resolver/TransactionLinkResolver.ts

```diff
@@ -0,0 +1,98 @@
+import { calculateDecay } from '@/util/decay'
+
+// TODO: do not export, test it inside the resolver
+export const transactionLinkCode = (date: Date): string => {
+  const time = date.getTime().toString(16)
+  return (
+    randomBytes(12)
+      .toString('hex')
+      .substring(0, 24 - time.length) + time
+  )
+}
+
+const CODE_VALID_DAYS_DURATION = 14
+
+const transactionLinkExpireDate = (date: Date): Date => {
+  const validUntil = new Date(date)
+  return new Date(validUntil.setDate(date.getDate() + CODE_VALID_DAYS_DURATION))
+}
+
+@Resolver()
+export class TransactionLinkResolver {
+  @Authorized([RIGHTS.CREATE_TRANSACTION_LINK])
+  @Mutation(() => TransactionLink)
+  async createTransactionLink(
+    @Args() { amount, memo, showEmail = false }: TransactionLinkArgs,
+    @Ctx() context: any,
+  ): Promise<TransactionLink> {
+    const userRepository = getCustomRepository(UserRepository)
+    const user = await userRepository.findByPubkeyHex(context.pubKey)
+
+    const createdDate = new Date()
+    const validUntil = transactionLinkExpireDate(createdDate)
+
+    const holdAvailableAmount = amount.minus(calculateDecay(amount, createdDate, validUntil).decay)
+
+    // validate amount
+    const sendBalance = await calculateBalance(user.id, holdAvailableAmount.mul(-1), createdDate)
+    if (!sendBalance) {
+      throw new Error("user hasn't enough GDD or amount is < 0")
+    }
+
+    const transactionLink = dbTransactionLink.create()
+    transactionLink.userId = user.id
+    transactionLink.amount = amount
+    transactionLink.memo = memo
+    transactionLink.holdAvailableAmount = holdAvailableAmount
+    transactionLink.code = transactionLinkCode(createdDate)
+    transactionLink.createdAt = createdDate
+    transactionLink.validUntil = validUntil
+    transactionLink.showEmail = showEmail
+    await dbTransactionLink.save(transactionLink).catch(() => {
+      throw new Error('Unable to save transaction link')
+    })
+
+    return new TransactionLink(transactionLink, new User(user))
+  }
+
+  @Authorized([RIGHTS.DELETE_TRANSACTION_LINK])
+  @Mutation(() => Date, { nullable: true })
+  async deleteTransactionLink(
+    @Arg('id') id: number,
+    @Ctx() context: any,
+  ): Promise<Date | null | undefined> {
+    const userRepository = getCustomRepository(UserRepository)
+    const user = await userRepository.findByPubkeyHex(context.pubKey)
+
+    const transactionLink = await dbTransactionLink.findOne({ id })
+    if (!transactionLink) {
+      throw new Error('Transaction Link not found!')
+    }
+
+    // TODO: admin can delete links?
+    if (transactionLink.userId !== user.id) {
+      throw new Error('Transaction Link cannot be deleted!')
+    }
+
+    if (transactionLink.redeemedBy) {
+      throw new Error('Transaction Link already redeemed!')
+    }
+
+    await transactionLink.softRemove()
+    const newLink = await dbTransactionLink.findOne({ id }, { withDeleted: true })
+    return newLink ? newLink.deletedAt : null
```

Comment:
how is this relevant for the frontend? Deleted links are not rendered at all

```suggestion
    return true
```